asm/macros/event.inc

```diff
@@ -251,9 +251,13 @@
 	.endm
 
 	@ Calls the native C function stored at func.
-	.macro callnative func:req
+	.macro callnative func:req, player_visible=TRUE
 	.byte 0x23
+	.if \player_visible
 	.4byte \func
+	.else
+	.4byte \func + 0x02000000
```

Comment:
What precisely is the function of this offset? Feels a bit like sorcery if you don't know the context, so I'd consider elaborating with a comment

Reply:
To explain the functionality, a `0x08xxxxxx`/`0x09xxxxxx` ROM address plus `0x02000000` produces a mirrored ROM address. You can load from that pointer and it references the same location as the original address, but we can now detect them. This same trick was used by merrp for the "mirrorred strings" in Guillotine.

I'm not sure if we'd want to have that explanation in the three places it's used (`callnative`, `defspecial`, and `MUT_INSTRUMENTED`), but I can do that if you think it makes sense.

Reply:
The functionality of mirrorring is way above my head, but maybe the 0x02000000 can become a define too? Not sure, but it does make it considerably less scary imo

Reply:
I would appreciate an explanation of when `player_visible` must be explicitly `FALSE`and when to wrap a script command in `MUT_INSTRUMENTED`. I assume it is when they dont have a player visible effect, but clarifying when exactly that is the case would help those who dont entirel understand what is happening here ~~totally not me btw~~. I see that you already explained this down below a bit already but a comment along the lines of: `player_visible` must be `FALSE` when .... `

Reply:
I'll leave comments which direct people to `Script_HasNoEffect`'s documentation to avoid repeating myself. Does the comment above that function give you confidence that you'd be able to make the appropriate changes, and if not do you have any suggestions on what could be more detailed, or what examples to give?

Reply:
maybe just adding a short sentence that a special or callnative function that doesnt mutate anything other than the listed variables must specify this.